extensions/security/runtime/src/main/java/io/quarkus/security/runtime/SecurityBuildTimeConfig.java

```diff
@@ -29,5 +29,9 @@
      */
     @ConfigItem(name = "deny-unannotated-members")
     public boolean denyUnannotated;
-
+    /**
+     * If set to true, security annotations will be inherited.
+     */
+    @ConfigItem(name = "inherit-security-annotations")
+    public boolean inheritSecurityAnnotations;
```

Comment:
I think if we are going to do this then should be true by default.

Reply:
Thanks Stuart, so you think we should merge this for now ? or directly implement the above as drafted by @sberyozkin @mkouba @manovotn ? I can have a look at arc implementation to see if I can help to reimplement it the way it is described above.

Reply:
done